packages/scene-composer/src/components/StateManager.tsx

```diff
@@ -319,11 +319,7 @@ const StateManager: React.FC<SceneComposerInternalProps> = ({
 
   useEffect(() => {
     if (onSceneLoaded && sceneLoaded && (!enableMatterportViewer || matterportReady)) {
-      // Delay the event handler to let other components finish loading, otherwise the consumer side will
-      // fail to update scene states
-      setTimeout(() => {
```

Comment:
These `setTimeout` functions were providing no functional capability & only impaired the event lifecycle. Progress bar is still completely functional to our current capabilities. 

https://github.com/user-attachments/assets/12b128cb-5737-4237-ad4c-7b6416cc25a7